### ISSUE TITLE
Version 8.0.0 does not work in browser

### ISSUE THREAD

**Issue by miyasudokoro** [chaijs/chai-as-promised#302]
I have been using Browserify for years to convert chai-as-promised for the client side, so I was overjoyed to see that version 8.0.0 is now in ESM. I have a "canary" unit test file used for version upgrading to ensure various libraries/plugins/etc. are interoperable. This 8.0.0 version of chai-as-promised does not pass. 

These tests are in Mocha / Chai / Sinon. Additional plugins in use are sinon-chai and dirty-chai. This failure occurs using Karma as well as @web/test-runner chromeLauncher and "manual" mode. 

```
    describe( 'chai-as-promised', () => {
        it( 'expects promises', () => {
            const promise = new Promise( ( resolve, _reject ) => {
                setTimeout( () => resolve( 'abc' ) );
            } );
            return expect( promise )
                .to.eventually.equal( 'abc' );
        } );
    } );
```

The failure is as follows:

```
1) canary
       chai-as-promised
         expects promises:
     AssertionError: expected Promise{…} to equal 'abc'
```

In other words, it's comparing the Promise instance rather than what the Promise instance resolves to.

While debugging it, I can see that there is a line `utils.flag(this, 'eventually', true);` that is reached during the test, but unfortunately, the conditional `if (!utils.flag(assertion, 'eventually')) {` is never reached after that. That's as far as I got while attempting to debug so far.

**Comment by 43081j** [comment]
Could you possibly make a small GitHub or gist to reproduce this?

It would us investigate it for sure

**Comment by miyasudokoro** [comment]
I have been debugging this a lot more this morning. I've narrowed it to the combination of plugins chai-as-promised and dirty-chai, which I have been doing for 6+ years without issue using browserified versions. Removing dirty-chai causes chai-as-promised to work again. I am going to keep investigating for a while, but maybe this is going to end up as a dirty-chai issue instead of a chai-as-promised issue. :)

If anyone wants to reproduce it, here you go. 

```lang=html
<!DOCTYPE html>
<html lang="en">
<head>
    <title>Canary test</title>
    <link rel="stylesheet" href="node_modules/mocha/mocha.css">
</head>
<body>
<div id="mocha"></div>
<script type="importmap">
    {
        "imports": {
            "check-error": "./node_modules/check-error/index.js"
        }
    }
</script>
<script type="module">
    import '/node_modules/mocha/mocha.js'; // window.mocha
    import '/node_modules/chai/chai.js'; // window.chai
    import '/node_modules/sinon-chai/lib/sinon-chai.js';
    // has chai.use( sinonChai ) inside

    import '/node_modules/dirty-chai/lib/dirty-chai.js';
    // has chai.use( dirtyChai ) inside
    
    import chaiAsPromised from '/node_modules/chai-as-promised/fuckyou/chai-as-promised.js';
    chai.use( chaiAsPromised );
    

    const { expect } = chai;

    mocha.setup( 'bdd' );

    describe( 'chai-as-promised', () => {
        it( 'expects promises', async() => {
            const promise = new Promise( ( resolve, _reject ) => {
                setTimeout( () => {
                    resolve( 'abc' )
                }, 1 );
            } );
            return expect( promise )
                .to.eventually.equal( 'abc' );
        } );
    } );

    mocha.run();
</script>
</body>
</html>


```

**Comment by miyasudokoro** [comment]
My conclusion is, dirty-chai has not been updated for 7 years, and it's no longer compatible with the latest versions of chai and chai-as-promised. I will either fork and update it myself or remove its use from my projects. I don't think there's any problem with chai-as-promised.
